Zend/zend_execute.c

```diff
@@ -873,14 +873,12 @@ ZEND_COLD zend_never_inline void zend_magic_get_property_type_inconsistency_erro
 ZEND_COLD void zend_match_unhandled_error(const zval *value)
 {
 	smart_str msg = {0};
-
-	if (Z_TYPE_P(value) <= IS_STRING) {
-		smart_str_append_scalar(&msg, value, EG(exception_string_param_max_len));
+	if (smart_str_append_zval(&msg, value, EG(exception_string_param_max_len)) == SUCCESS) {
+		/* Nothing to do. */
 	} else {
```

Comment:
Maybe?
```suggestion
	if (smart_str_append_zval(&msg, value, EG(exception_string_param_max_len)) == FAILURE) {
```

Reply:
Nope, that looks correct. We only want to emit "of type ..." if `smart_str_append_zval()` can't handle the value. Note that it intentionally does not handle some values like arrays, because they seem to be handled differently depending on the context. It would be nice if it could be unified in the future, but my intention is not to break BC here.

Reply:
Oh, sorry. I misread your suggestion. :) Yeah, I'll change it!